events/moose/moose.go

```diff
@@ -306,19 +306,20 @@ func (s *Subscriber) NotifyConnect(data events.DataConnect) error {
 		return nil
 	}
 
-	s.connectedAt = time.Now()
-	dnsResolutionTime := int(data.DNSResolutionTime.Milliseconds())
 	var threatProtection moose.Enum_SS_NordvpnappOptBool
 	if data.ThreatProtectionLite {
 		threatProtection = moose.Enum_SS_NordvpnappOptBool(moose.OptBoolTrue)
 	} else {
 		threatProtection = moose.Enum_SS_NordvpnappOptBool(moose.OptBoolFalse)
 	}
 
+	eventDuration := int(time.Since(s.connectedAt).Seconds())
```

Comment:
Maybe initialize here to -1, and then overwrite with actual calculated values in `case events.ConnectSuccess:` and `case events.ConnectFailure:`? The outcome will be the same, but it would be more logical and will also send -1 instead some incorrect value in `default:` case.